data_steward/cdr_cleaner/cleaning_rules/deid/conflicting_hpo_state_generalization.py

```diff
@@ -0,0 +1,250 @@
+  FROM `{{project_id}}.{{sandbox_dataset_id}}.{{target_table}}` AS SB_OBS
+  WHERE D_OBS.observation_id = SB_OBS.observation_id
+""")
+
+
+class ConflictingHpoStateGeneralize(BaseCleaningRule):
+    """Generalize conflicting hpo states."""
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id,
+                 table_namer=None):
+        desc = (
+            "Set all participant home states to a generalized value if they have "
+            "EHR records from a different state.")
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         issue_urls=JIRA_ISSUE_URL,
+                         description=desc,
+                         affected_datasets=[cdr_consts.REGISTERED_TIER_DEID],
+                         affected_tables=[OBSERVATION],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         table_namer=table_namer)
+
+    def setup_rule(self, client, *args, **keyword_args):
+        """
+        Run required steps for setup rule
+
+        1. Identify the states where EHR data is originating from.
+        2. Identify the states participants claim to live in
+           (This is value_source_concept_id where observation_source_concept_id = 1585249.)
+        3. If you find any EHR records for a participant that originates from a state other than the state
+           where the participant claims to live, then that participants original response should be sandboxed
+           and their data should be generalized such that value_source_concept_id = 2000000011 and
+           value_as_concept_id = 2000000011  when observation_source_concept_id = 1585249.
+        """
+        # Currently this rule is only run for 'OBSERVATION' table.
+        # To include more tables in the future, update 'affected_tables' in __init__ args.
+
+        # List Dataset Content
+        dataset_tables = client.list_tables(self.dataset_id)
+        dataset_table_ids = [table.table_id for table in dataset_tables]
+
+        mapped_tables = [
+            table[9:]
+            for table in dataset_table_ids
+            if table.startswith('_mapping_')
+        ]
+
+        # Make sure all mapped_tables exists
+        check_tables = [
+            table for table in mapped_tables if table in dataset_table_ids
+        ]
+
+        # Make sure check_tables contains person_id field
+        person_id_query = LIST_PERSON_ID_TABLES.render(
+            project_id=self.project_id, dataset_id=self.dataset_id)
+        result = client.query(person_id_query).result()
+
+        # List of mapped tables that exists in given dataset with 'person_id' column.
+        person_id_tables = []
+        for row in result:
+            table = row.get('table_name')
+            if table in check_tables:
+                person_id_tables.append(table)
+
```

Comment:
I think you can accomplish all this with a much simpler check now.  This entire thing was attempting to find all tables with a `person_id` column that also had a `_mapping_<omop table>`.  

I think you could do the same thing with a query now and remove a lot of this old code.
```sql  
  SELECT
  DISTINCT table_name
  FROM `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS`
  WHERE lower(column_name) = 'person_id'
  and table_name || "_ext" in (
    select distinct table_id 
    from `{{project}}.{{dataset}}.__TABLES__` 
    where REGEXP_CONTAINS(table_id, r'(?i)_ext$')
)
```